py-blis: fix environment settings

QianJianhua1 opened the pull request:
Error occur when compiling with a compiler other than GCC, or on a platform other than x86_64. The reason is that two environment variables are not set.
-  BLIS_COMPILER
-  BLIS_ARCH

QianJianhua1 committed:
py-blis: fix environment settings

adamjstewart commented on var/spack/repos/builtin/packages/py-blis/package.py:
```diff
@@ -18,3 +18,7 @@ class PyBlis(PythonPackage):
     depends_on("py-numpy@1.15:", type=("build", "run"))
+
+    def setup_build_environment(self, env):
+        env.set("BLIS_COMPILER", self.compiler.cc)
```
Is this the compiler used at build time or the compiler used at run time? If the former, you'll want to use:
```suggestion
        env.set("BLIS_COMPILER", spack_cc)
```
so that it uses Spack's compiler wrappers.

QianJianhua1 committed:
Update var/spack/repos/builtin/packages/py-blis/package.py

Thank you. This compiler used at build time. I have tested with 'spack_cc', it's OK.

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>
